esp32/littlefs/vfs_littlefs_file.c

```diff
@@ -100,17 +100,19 @@ STATIC mp_uint_t file_obj_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_t arg,
         return 0;
 
     } else if (request == MP_STREAM_CLOSE) {
+        if (self->littlefs == NULL) {
+            return 0;
+        }
 
         xSemaphoreTake(self->littlefs->mutex, portMAX_DELAY);
             int res = littlefs_close_common_helper(&self->littlefs->lfs, &self->fp, &self->cfg, &self->timestamp_update);
         xSemaphoreGive(self->littlefs->mutex);
+
+        self->littlefs = NULL; // indicate a closed file
```

Comment:
I think this should be moved after checking the return value, because if the return value is smaller than 0 then an error occured which means the file has not been closed. I would move this to line 111.